cairosvg/url.py

```diff
@@ -32,6 +32,64 @@
 URL = re.compile(r'url\((.+)\)')
 
 
+class URLFetcher(object):
```

Comment:
Update in new PR.

With this update it is still relative simpel to create a URLFetcher which will only allow resources beneath the base url of the initial SVG. For example:

    class BaseURLFetcher(URLFetcher):

        def __init__(self):
            super().__init__()
            self.base = None

        def fetch(self, url):
            # Check for external resources (ie scheme is not data:)
            parsed_url = parse_url(url, self.base)
            if parsed_url.scheme != 'data':
                # If no base is registered yet, this first URL provides base.
                # Otherwise check if base of URL equals the previously stored base.
                url_base = os.path.dirname(url) + '/'
                if self.base is None:
                    self.base = url_base
                elif not url_base.startswith(self.base):
                    raise Exception("URL outside base not accepted")
            return super().fetch(url)